### PR TITLE
Implement improvements for sheets/charts integration

### PR DESCRIPTION
**Pull request by ewpatton**
This PR includes improvements for the Spreadsheet and Chart components based on curriculum usability feedback from @bobparks. The changes are available for testing at http://pr2786.ai2-ewpatton-temp.appspot.com/. The PR is broken down into a sequence of smaller commits that can be reviewed independently.

### CODE REVIEW

File: appinventor/blocklyeditor/src/versioning.js [hunk 1]
```diff
@@ -2982,6 +2983,22 @@ Blockly.Versioning.AllUpgradeMaps =
...
+      Blockly.Versioning.changeMethodName("Spreadsheet", "WriteCol", "WriteColumn"),
+      Blockly.Versioning.changeMethodName("Spreadsheet", "AddCol", "AddColumn"),
+      Blockly.Versioning.changeMethodName("Spreadsheet", "RemoveCol", "RemoveColumn"),
+      Blockly.Versioning.changeEventName("Spreadsheet", "GotColData", "GotColumnData"),
+      Blockly.Versioning.changeEventName("Spreadsheet", "FinishedWriteCol", "FinishedWriteColumn"),
+      Blockly.Versioning.changeEventName("Spreadsheet", "FinishedAddCol", "FinishedAddColumn"),
+      Blockly.Versioning.changeEventName("Spreadsheet", "FinishedRemoveCol", "FinishedRemoveColumn"),
+      Blockly.Versioning.changeEventParameterName("Spreadsheet", "GotFilterResult", "return_rows", "returnRows"),
+      Blockly.Versioning.changeEventParameterName("Spreadsheet", "GotFilterResult", "return_data", "returnData"),
+      Blockly.Versioning.changeEventParameterName("Spreadsheet", "GotColumnData", "colDataList", "columnData")
```
Comment (SusanRatiLane) [inline appinventor/blocklyeditor/src/versioning.js:2998]: Is the migration intended to handle user code as well as the event itself? I imported an app ABG uploaded, and while the event GotFilterResult's return value reads returnData, his code is still trying to access return_data and throwing an error.

File: appinventor/blocklyeditor/src/versioning.js [hunk 2]
```diff
@@ -2982,6 +2983,22 @@ Blockly.Versioning.AllUpgradeMaps =
 
   }, // End Spinner upgraders
 
+  "Spreadsheet": {
+    2: [
+      Blockly.Versioning.changeMethodName("Spreadsheet", "ReadCol", "ReadColumn"),
+      Blockly.Versioning.changeMethodName("Spreadsheet", "WriteCol", "WriteColumn"),
+      Blockly.Versioning.changeMethodName("Spreadsheet", "AddCol", "AddColumn"),
+      Blockly.Versioning.changeMethodName("Spreadsheet", "RemoveCol", "RemoveColumn"),
+      Blockly.Versioning.changeEventName("Spreadsheet", "GotColData", "GotColumnData"),
```
Comment (SusanRatiLane) [inline appinventor/blocklyeditor/src/versioning.js:2992]: We talked about changing the names of these events to better match the naming conventions that appear elsewhere in App Inventor, which in turn better matches industry standard. Do we want to go ahead and do that here? E.g. ColumnWritten (WroteColumn?), ColumnAdded, etc. This looks more in line with what I see in TinyWebDB and CloudDB.

File: appinventor/blocklyeditor/src/versioning.js (hunk 1 above)
Comment (ewpatton) [reply appinventor/blocklyeditor/src/versioning.js:3006]: This should be fixed with the latest commit.

File: appinventor/blocklyeditor/src/versioning.js (hunk 2 above)
Comment (ewpatton) [reply appinventor/blocklyeditor/src/versioning.js:3000]: I'm not sure that we want to hold up the release for that.

**Review by SusanRatiLane — APPROVED** [review]
Looks good. Migration update checks out.

**Pull request merged**